### DISCUSSION TITLE
How to input an array parameter in query?

### DISCUSSION THREAD

**Discussion by bsdayo** [scalar/scalar#2835] [Q&A]
I have the following OpenAPI spec:

```json
{
  "openapi": "3.0.1",
  "info": {
    "title": "example",
    "version": "1.0"
  },
  "paths": {
    "/announcements": {
      "get": {
        "parameters": [
          {
            "name": "dismiss",
            "in": "query",
            "schema": {
              "type": "array",
              "items": {
                "type": "integer",
                "format": "int32"
              }
            }
          }
        ],
        "responses": {
          "200": {
            "description": "OK"
          }
        }
      }
    }
  }
}
```

In Swagger I can do something like this:

![image](https://github.com/user-attachments/assets/403fefe2-e0c0-44a9-bfed-3dd52a1dc4c2)

Results in `?dismiss=123&dismiss=456`, which is a desired behavior.

But in Scalar, the parameter is displayed as a simple text field, which cannot be splited into multiple items:

![image](https://github.com/user-attachments/assets/b65f9ef5-6e37-44d9-a954-47b9f54ce95d)

I have tried:

- `123,456`
- `123 456`
- `123;456`

None of these approaches works. I have also tried creating 2 parameter with the same name `dismiss`, but the latter overrided the former.

Then how can I achieve the same result just like in Swagger? Thanks.

**Comment by hanspagel** [comment]
Interesting! We don’t handle this yet. Thanks for letting us know.

**Reply by bsdayo** [reply]
Could this feature be added to the roadmap? Scalar is a nice project that we cannot wait to migrate from Swagger and make use of it, but lacking of this feature is the main reason which prevents us from migrating (we use array parameters in query in many many APIs). Thanks!

**Comment by bfg-coding** [comment]
This is a feature I'm also heavily reliant on.  Is there an "issue" to keep track on adding it to the project?

**Comment by MarvinKlein1508** [comment]
I'm giving my vote to this.

**Comment by MarvinKlein1508** [comment]
I've created an issue to track this.
https://github.com/scalar/scalar/issues/4232

**Comment by antlio** [comment]
thank you all for reporting this and the precious feedbacks! we'll start working on this soon!
